src/main/java/com/iexec/worker/replicate/ReplicateDemandService.java

```diff
@@ -67,11 +68,12 @@ public ReplicateDemandService(IexecHubService iexecHubService,
      * We use single thread executor to make sure the worker does not ask for more
      * than one replicate at the same time. The executor's queue is of size 1 to
      * avoid memory leak if the thread halts for any reason.
+     * @return A {@link CompletableFuture} representing the executing task.
      */
-    @Scheduled(fixedRateString = "#{publicConfigurationService.askForReplicatePeriod}")
-    void triggerAskForReplicate() {
+    @Scheduled(fixedDelayString = "#{publicConfigurationService.askForReplicatePeriod}")
+    CompletableFuture<Void> triggerAskForReplicate() {
```

Comment:
I guess `CompletableFuture<Void>` is added for testing purposes? I don't mind keeping it this way although I would've preferred to sleep for some milliseconds in the test rather than modifying the method.

Reply:
Your guess is right.
Sleeping in test is generally a bad idea, see https://rules.sonarsource.com/java/RSPEC-2925.